solr/core/src/java/org/apache/solr/index/SlowCompositeReaderWrapper.java

```diff
@@ -328,6 +331,17 @@ public Fields getTermVectors(int docID) throws IOException {
     return in.getTermVectors(docID);
   }
 
+  @Override
+  public TermVectors termVectors() throws IOException {
+    return in.termVectors();
```

Comment:
I know this follows precedent from the existing `termVectors()` method above, but I wonder why these don't call `ensureOpen()` the wall all the other "content-access" methods seem to do?

Reply:
Yeah, all the other functions seem to have `ensureOpen()` apart from `getTermVectors(int docID)` and `termVectors()` (of course I added this one).
Should I add for both functions or should I add only for `termVectors()` and will take care of `getTermVectors(int docID)` in some other PR?

Reply:
`ensureOpen()` was removed as part of [this commit](https://github.com/apache/solr/commit/797bb38c50fed9e533ca20de920c63c6c35256a9#diff-a3934253a307f6bc36771c8d549e5b5bb7bad12fd5dbe4de06e7b5ef176e2083).

Looks like an oversight perhaps. I'd be inclined to put it back (for consistency with the other methods). @markrmiller any recollection why this was removed? Any objection to restoring it?

Reply:
Hey @magibney, should we wait for his reply or should we add it back?

Reply:
I'd recommend to go ahead and add it back. Thanks for following up!